modules/solid_mechanics/include/userobjects/CrackFrontDefinition.h

```diff
@@ -133,6 +137,13 @@ class CrackFrontDefinition :
                                                const RealVectorValue& tangent_direction,
                                                const CRACK_NODE_TYPE ntype) const;
   void calculateTangentialStrainAlongFront();
+  void createQFunctionRings();
+  void addNodesToQFunctionRing(std::set<dof_id_type>& nodes_new_ring,
+                               const std::set<dof_id_type> nodes_old_ring,
```

Comment:
Did you mean to pass all these std::sets by constant reference?  const parameters passed by value do have their uses, but you are still making a copy of these...